src/opl3.c

```diff
@@ -1160,17 +1158,17 @@ void OPL3_Generate(opl3_chip *chip, Bit16s *buf)
         OPL3_SlotGenerate(&chip->slot[ii]);
     }
 
-    chip->mixbuff[1] = 0;
+    mixed = 0;
     for (ii = 0; ii < 18; ii++)
     {
-        accm = 0;
-        for (jj = 0; jj < 4; jj++)
-        {
-            accm += *chip->channel[ii].out[jj];
-        }
-        chip->mixbuff[1] += (Bit16s)(accm & chip->channel[ii].chb);
+        const opl3_channel *chan = &chip->channel[ii];
+        Bit16s * const *chanout = chan->out;
+        const Bit16s accm = *chanout[0] + *chanout[1] + *chanout[2] + *chanout[3];
+        mixed += (Bit16s)(accm & chan->cha);
```

Comment:
should be chan->chb

Reply:
whoops, copy/paste error, sorry about that! Pushing a fix.